CRM/Core/BAO/File.php

```diff
@@ -689,10 +689,30 @@ public static function deleteAttachment() {
       //     ],
       //   ]);
       // TEMP HACK
-      CRM_Core_DAO::executeQuery("UPDATE `{$customGroup['table_name']}` SET {$customField['column_name']} = NULL WHERE entity_id = %1 AND {$customField['column_name']} = %2", [
+      $ids = CRM_Core_DAO::executeQuery("SELECT id FROM `{$customGroup['table_name']}` WHERE entity_id = %1 AND {$customField['column_name']} = %2", [
         1 => [$params['entityID'], 'Integer'],
         2 => [$params['fileID'], 'Integer'],
       ]);
+      $customParams = [$customGroup['table_name'] => []];
+      while ($ids->fetch()) {
+        $customParams[$customGroup['table_name']][$dao->id] = [];
```

Comment:
Where does `$dao->id` come from? The `$dao` variable does not seem to be defined in this function. Does it contain the id from the query? How does it end up in here? Shouldn't the result of `$ids->fetch()` be captured?

Reply:
@seamuslee001 pls check this comment I think @Sjord is correct.

Reply:
thanks @Sjord yes this was a typo